### PR TITLE
Simplify TypeKind condition

### PR DESCRIPTION
**Pull request by Youssef1313**
`TypeKind.Struct` and `TypeKind.Structure` have the same value.

### CODE REVIEW

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/Core/Portable/GenerateType/AbstractGenerateTypeService.GenerateNamedType.cs [hunk 1]
```diff
@@ -216,7 +216,7 @@ private void AddProperties(ArrayBuilder<ISymbol> members)
                 }
 
                 // Empty Constructor for Struct is not allowed
-                if (!(parameters.Count == 0 && options != null && (options.TypeKind == TypeKind.Struct || options.TypeKind == TypeKind.Structure)))
+                if (!(parameters.Count == 0 && options != null && options.TypeKind == TypeKind.Struct))
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/GenerateType/AbstractGenerateTypeService.GenerateNamedType.cs:219]:
```suggestion
                if (!(parameters.Count == 0 && options is { TypeKind: TypeKind.Struct })
```

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/GenerateType/AbstractGenerateTypeService.GenerateNamedType.cs:219]: or `?.TypeKind == TypeKind.Struct`

File: src/Features/Core/Portable/GenerateType/AbstractGenerateTypeService.GenerateNamedType.cs [hunk 2]
```diff
@@ -216,7 +216,7 @@ private void AddProperties(ArrayBuilder<ISymbol> members)
                 }
 
                 // Empty Constructor for Struct is not allowed
-                if (!(parameters.Count == 0 && options != null && (options.TypeKind == TypeKind.Struct || options.TypeKind == TypeKind.Structure)))
+                if (!(parameters.Count == 0 && options is { TypeKind: TypeKind.Struct })
```
Comment (Youssef1313) [inline src/Features/Core/Portable/GenerateType/AbstractGenerateTypeService.GenerateNamedType.cs:219]:
```suggestion
                if (!(parameters.Count == 0 && options is { TypeKind: TypeKind.Struct }))
```

**Comment by Youssef1313** [comment]
@CyrusNajmabadi This is ready to merge.

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**